### PR TITLE
NoMultilineWhitespaceBeforeSemicolonsFixer - Semicolon should not be moved into comment

### PR DESCRIPTION
**Pull request by SpacePossum**
Bug on master only. 
The fixer is broken because of the transformer that splits up a comment with a trailing line break.

Closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1808

### CODE REVIEW

**Comment by keradus** [comment]
Please fix CS

**Comment by SpacePossum** [comment]
done :)

File: src/Fixer/Semicolon/NoMultilineWhitespaceBeforeSemicolonsFixer.php [hunk 1]
```diff
@@ -39,8 +39,13 @@ public function fix(\SplFileInfo $file, Tokens $tokens)
             }
 
             $previous = $tokens[$index - 1];
+            if (!$previous->isWhitespace() || false === strpos($previous->getContent(), "\n")) {
+                continue;
+            }
 
-            if ($previous->isWhitespace() && false !== strpos($previous->getContent(), "\n")) {
+            if (("\n" === $previous->getContent()[0] || "\r" === $previous->getContent()[0]) && $tokens[$index - 2]->isComment()) {
+                $previous->setContent("\r" === $previous->getContent()[0] ? "\r\n" : "\n");
```
Comment (GrahamCampbell) [inline src/Fixer/Semicolon/NoMultilineWhitespaceBeforeSemicolonsFixer.php]: php 5.3 doesn't support array access on the result of a function call

Comment (GrahamCampbell) [inline src/Fixer/Semicolon/NoMultilineWhitespaceBeforeSemicolonsFixer.php]: you'll need to assign it to variable first

Comment (keradus) [inline src/Fixer/Semicolon/NoMultilineWhitespaceBeforeSemicolonsFixer.php]: Or just call `substr`

Comment (GrahamCampbell) [inline src/Fixer/Semicolon/NoMultilineWhitespaceBeforeSemicolonsFixer.php]: with long array syntax, and a true as the 3rd param :trollface:

Comment (keradus) [inline src/Fixer/Semicolon/NoMultilineWhitespaceBeforeSemicolonsFixer.php]: I didn't want to write the code for him ;P

Comment (SpacePossum) [inline src/Fixer/Semicolon/NoMultilineWhitespaceBeforeSemicolonsFixer.php]: lol, no :P

**Comment by keradus** [comment]
@SpacePossum ?

**Comment by SpacePossum** [comment]
IMHO this fixer shouldn't be on master in the first place. Using `spaces_before_semicolon` and `extra_empty_lines` will give better results. So I lost my interest beyond the point going through another round of reviews.

**Comment by keradus** [comment]
Only concerns here were to not break fabpot / PHP 5.3. So nothing big.
So you are saying we don't need this fixer at all?

**Comment by SpacePossum** [comment]
I think this fixer should be merged with `spaces_after_semicolon`, possible with some config options if one really cares, however I don't have the time.
Test are green now :)

PS. This stops the fixer from breaking stuff, however it would be nice to be tweaked at some point to do;
expected
```php
<?php
echo $a; // test
```
input
```php
<?php
echo $a // test

;
```

currently fixed to
```php
<?php
echo $a // test
;
```

**Comment by keradus** [comment]
It's good enough then. But I don't think it will be done soon (combining with `spaces_after_semicolon`) as well.

I think it's ready to merge.

**Comment by gharlan** [comment]
:+1:

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Comment by SpacePossum** [comment]
thanks all :)

**Pull request closed without merging**